src/OmniSharp.Host/CompositionHostBuilder.cs

```diff
@@ -49,15 +50,19 @@ public CompositionHost Build()
             var assemblyLoader = _serviceProvider.GetRequiredService<IAssemblyLoader>();
             var config = new ContainerConfiguration();
 
-            var assemblies = _assemblies
-                .Concat(new[] { typeof(OmniSharpWorkspace).GetTypeInfo().Assembly, typeof(IRequest).GetTypeInfo().Assembly })
-                .Distinct();
-
-            config = config.WithAssemblies(assemblies);
-
             var fileSystemWatcher = new ManualFileSystemWatcher();
             var metadataHelper = new MetadataHelper(assemblyLoader);
 
+            // We must register an MSBuild instance before composing MEF to ensure that
+            // our AssemblyResolve event is hooked up first.
+            var msbuildLocator = _serviceProvider.GetRequiredService<IMSBuildLocator>();
+            var instances = msbuildLocator.GetInstances();
+            var instance = instances.FirstOrDefault();
+            if (instance != null)
```

Comment:
Are we supposed to do anything if we couldn't locate any instances?

Reply:
We should probably report an error to the log. Will add that.